templates/console/source/lambda/config-version/cf_config_version_manager_graphql/cf_config_version_manager_graphql.py

```diff
@@ -694,23 +726,56 @@
             'dateTime': current_time,
             'note': snapShotNote,
         })
-
     return {
         'statusCode': 200,
         'body': 'succeed create new snapshot'
     }
 
 
+def get_latest_version(ddb_client, distributionId):
+    # first get the latest versionId of updated distribution
+    latest_table = ddb_client.Table(DDB_LATESTVERSION_TABLE_NAME)
+    try:
+        resp = latest_table.query(
+            KeyConditionExpression=Key('distributionId').eq(distributionId)
+        )
+        log.info(resp)
+    except Exception as e:
+        logging.error(e)
+    record_list = resp['Items']
+    if len(record_list) == 0:
+        raise Exception("There is no latest version for distributionId:" + distributionId)
+    latest_version = record_list[0]['versionId']
+    logging.info("The latest version of distribution:" + str(distributionId) + " is " + str(latest_version))
+    return latest_version
+
+
+def check_snapshot_name_duplication(ddb_table, distributionId, snapShotName):
+    response = ddb_table.get_item(
+        Key={
+            "distributionId": distributionId,
+            "snapShotName": snapShotName
+        })
+    logger.info(response)
+    if 'Item' in response:
+        # No duplicate snapShotName allowed
+        raise Exception("There is already snapShotName:" + snapShotName)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.